hedera-node/src/main/java/com/hedera/services/txns/crypto/AutoCreationLogic.java

```diff
@@ -165,28 +168,37 @@ public void submitRecordsTo(final RecordsHistorian recordsHistorian) {
      *
      * @param change a triggering change with unique alias
      * @param accountsLedger the accounts ledger to use for the provisional creation
+     * @param changes list of all changes need to construct tokenAliasMap
      * @return the fee charged for the auto-creation if ok, a failure reason otherwise
      */
     public Pair<ResponseCodeEnum, Long> create(
             final BalanceChange change,
-            final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger) {
+            final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger,
+            final List<BalanceChange> changes) {
         if (!usageLimits.areCreatableAccounts(1)) {
             return Pair.of(MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED, 0L);
         }
-        final var alias = change.alias();
+
+        final ByteString alias = getAlias(change);
```

Comment:
Do we still need the `getAlias()` private helper here, or can we just
```
final var alias = change.getAliasIfPresent();
if (alias == null) {
    throw new IllegalStateException("Cannot auto-create an account from unaliased change " + change);
}
```
?

Reply:
Yes, we can use the `getNonEmptyAliasIfPresent` method. But we don't need to throw if its empty as we check its not empty before entering `autoCreationLogic` right ?